core/templates/dev/head/pages/exploration-editor-page/services/learner-answer-details-data.service.ts

```diff
@@ -36,12 +36,19 @@ angular.module('oppia').factory('LearnerAnswerDetailsDataService', [
       '<entity_id>');
 
     var _fetchLearnerAnswerInfoData = function() {
-      var learnerAnswerInfoDataUrl = UrlInterpolationService.interpolateUrl(
-        LEARNER_ANSWER_INFO_DATA_URL, {
-          entity_type: 'exploration',
-          entity_id: _expId});
-
-      return $http.get(learnerAnswerInfoDataUrl);
+      return $http.get(
+        '/userinfohandler'
+      ).then(function(response) {
+        if (response.data.user_is_logged_in) {
```

Comment:
I think we want this to work if the user has edit rights, isn't it?

Have you tried testing this with a user who doesn't have the rights to edit the exploration?

Reply:
I am sorry I do not get you. If the user is not logged in, we cannot check any rights of the user, right? If the user is logged in the handler will be called as usual and will check the rights as expected.

Reply:
```
Create 2 accounts (A & B):

A creates and published the exploration and B visits the exploration editor page.
```

Have you tried testing the above scenerio^?

Reply:
Yes, the user B will get an error message:

<img width="1680" alt="Screenshot 2019-10-15 at 11 54 56 PM" src="https://user-images.githubusercontent.com/15226041/66858595-3a684680-efa7-11e9-89fc-a97fefa2ff89.png">

Reply:
So that's not the expected behavior as we allow any user to visit the published exploration editor page!

Reply:
Oh Ok, I got it. I have made the changes to fix it. I am using Editability service now to decide whether the user can see the answer info data or not.